vignettes/tcplfit2-vignette.Rmd

```diff
@@ -232,89 +269,95 @@ dat[wllt == "t" & acid %in% nconc[nconc > 1, acid],
 
-Once the data is initialized to a point where the required fields are available, the methods included in the `tcpl` package can be identified and applied without the need for a database connection. You can see the list of available methods for Level 3 in the table below:
+$$
+\% \space control = \frac{cval - bval}{pval - bval}
+$$
 
-```{r example4_mthdlist, fig.height = 6, fig.width = 7, warning = FALSE}
-mthd_funcs <- tcpl:::mc3_mthds()
+The baseline (`bval`) values are calculated as the medians of all data in neutral control wells by plate and the negative control (`pval`) value is calculated as the plate-wise minimum, by assay plate ID (`apid`), of the medians of corrected value (`cval`) of negative control wells.
+
+```{r example3_mthdlist, fig.height = 6, fig.width = 7, warning = FALSE}
+## Check out all level-3 methods for normalization in tcpl
 DT::datatable(tcpl::tcplMthdList(3),rownames= FALSE, options = list(scrollX = T))
 ```
 
-### Normalization
-Here three normalization methods are selected and applied to the data.  Note because of the way `tcpl` handles the application of functions, the dataframe must be called `dat`. In the future, `tcpl` will export these functions so that they can be applied to any dataset without the need for a specific name or dummy aeid.
+```{r example3 normalize}
+# calculate bval
+dat[, bval := median(cval[wllt == "n"]), by = list(apid)]
+# calculate pval
+dat[, pval := median(cval[wllt %in% c("m","o")], na.rm = TRUE), by = list(apid, wllt, conc)]
+dat[, pval := min(pval, na.rm = TRUE), by = list(apid)]
 
-```{r example4_mc3methods, fig.height = 6, fig.width = 7, results = 'hide'}
-# apply level 3 methods
-## These methods directly apply the normalization methods from tcpl without the need for a DB connection
-lapply(mthd_funcs[["bval.apid.nwlls.med"]](dummy_aeid), eval)
-lapply(mthd_funcs[["pval.apid.medncbyconc.min"]](dummy_aeid),eval)
-lapply(mthd_funcs[["resp.pc"]](dummy_aeid),eval)
+# Calculate normalized responses
+dat[, resp := ((cval - bval)/(pval - bval) * 100)]
 ```
 
-Level 4 determines the baseline variability, or noise, that will later be used for cutoff calculation. Using the established concentration index, the level 4 methods can be loaded in a similar way to level 3.
+Before model fitting, we need to determine the median absolute deviation around baseline (`BMAD`) and baseline variability (`onesd`), which are later used for cutoff and benchmark response (`BMR`) calculations, respectively. This is part of level-4 processing in `tcpl`. `tcplfit2` assumes no effect occurs in the two lowest concentrations. Thus, we consider test wells in the two lowest concentrations as our baseline to calculate `BMAD` and `onesd`.
 
-```{r example4_mthdlist_l4, fig.height = 6, fig.width = 7}
-mthd_funcs_l4 <- tcpl:::mc4_mthds()
+`BMD` can be calculated as the median absolute deviation of the data in control wells too. Check out other methods of determining `BMAD` and `onesd` used in `tcpl`.
```

Comment:
Do you really mean BMD here? Please clarify this section of text.

Reply:
bmad and onesd are both set at level 4. Note these two methods will be consolidated into one in tcpl v3.2